lib/scholar/neighbors/nn_descent.ex

```diff
@@ -0,0 +1,838 @@
+defmodule Scholar.Neighbors.NNDescent do
+  @moduledoc """
+  Nearest Neighbors Descent (NND) is an algorithm that calculates Approximated Nearest Neighbors (ANN)
+  for a given set of points[1].
+
+  It is implemented using Random Projection Forest[2].
+
+  ## References
+
+    [1] [Efficient K-Nearest Neighbor Graph Construction for Generic Similarity Measures](https://www.cs.princeton.edu/cass/papers/www11.pdf).
+    [2] Random projection trees and low dimensional manifolds
```

Comment:
Maybe replace this one with "Randomized partition trees for nearest neighbor search" (https://cseweb.ucsd.edu/~dasgupta/papers/exactnn-algo.pdf). Both papers are dealing with random projection trees, but the latter is focusing on k-NN.